src/components/dateTimePicker/index.tsx

```diff
@@ -108,79 +108,66 @@ export interface DateTimePickerProps {
   testID?: string;
 }
 
-interface DateTimePickerState {
-  prevValue?: Date;
-  value?: Date;
-}
-
 type DateTimePickerPropsInternal = DateTimePickerProps & BaseComponentInjectedProps;
 
-class DateTimePicker extends Component<DateTimePickerPropsInternal, DateTimePickerState> {
-  static displayName = 'DateTimePicker';
-
-  static defaultProps = {
-    ...TextField.defaultProps,
-    mode: MODES.DATE
-  };
-
-  chosenDate?: Date;
-  expandable = React.createRef<ExpandableOverlayMethods>();
+function DateTimePicker(props: DateTimePickerPropsInternal) {
+  const {
+    renderInput,
+    editable,
+    mode = MODES.DATE,
+    dateFormat,
+    timeFormat,
+    dateFormatter,
+    timeFormatter,
+    minimumDate,
+    maximumDate,
+    locale,
+    is24Hour,
+    minuteInterval,
+    timeZoneOffsetInMinutes,
+    themeVariant,
+    onChange,
+    dialogProps,
+    headerStyle,
+    // @ts-expect-error
+    useCustomTheme,
+    testID,
+    ...others
+  } = props;
 
-  constructor(props: DateTimePickerPropsInternal) {
-    super(props);
-    this.chosenDate = props.value;
+  const [value, setValue] = useState(props.value);
+  const chosenDate = useRef(props.value);
+  const expandable = useRef<ExpandableOverlayMethods>();
 
+  useEffect(() => {
     if (!RNDateTimePicker) {
       console.error(`RNUILib DateTimePicker component requires installing "@react-native-community/datetimepicker" dependency`);
     }
-  }
-
-  state = {
-    prevValue: this.props.value,
-    value: this.props.value
-  };
-
-  static getDerivedStateFromProps(nextProps: DateTimePickerProps, prevState: DateTimePickerState) {
-    if (nextProps.value !== prevState.prevValue) {
-      return {
-        prevValue: prevState.value,
-        value: nextProps.value
-      };
-    }
-    return null;
-  }
-
-  handleChange = (event: any = {}, date: Date) => {
-    // NOTE: will be called on Android even when there was no actual change
-    if (event.type !== 'dismissed' && date !== undefined) {
-      this.chosenDate = date;
-
-      if (Constants.isAndroid) {
-        this.onDonePressed();
-      }
-    } else if (event.type === 'dismissed' && Constants.isAndroid) {
-      this.toggleExpandableOverlay();
-    }
-  };
+  }, []);
 
-  toggleExpandableOverlay = () => {
-    this.expandable.current?.toggleExpandable?.();
-  };
-
-  onDonePressed = () => {
-    this.toggleExpandableOverlay();
-    if (Constants.isIOS && !this.chosenDate) {
-      // since handleChange() is not called on iOS when there is no actual change
-      this.chosenDate = new Date();
-    }
+  useEffect(() => {
```

Comment:
This useEffect will be called in mount as well, which is unnecessary as you already set it in the useState statement. I suggest using `useDidUpdate` hook instead.

Reply:
Right! Fixed!